frame/treasury/src/lib.rs

```diff
@@ -222,11 +274,11 @@ decl_module! {
 /// A spending proposal.
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Encode, Decode, Clone, PartialEq, Eq, sp_runtime::RuntimeDebug)]
-pub struct Proposal<AccountId, RingBalance> {
+pub struct Proposal<AccountId, StakingBalance> {
```

Comment:
I think Proposal can be something like Ledger @HackFisher 

```rust
pub struct Proposal<AccountId, RingBalance, KtonBalance> {
	proposer: AccountId,
	proposing_ring: RingBalance,
	proposing_kton: KtonBalance,
	bond_ring: RingBalance,
	bond_kton: KtonBalance,
}
```

Also good for merge `spend_kton_funds` and `spend_ring_funds`.

Reply:
It is a good idea, can help support more powerful combined proposal which can use RING and KTON the same time.
```
pub struct Proposal<AccountId, RingBalance> {
	proposer: AccountId,
        beneficiary: AccountId,
	ringValue: RingBalance,
	ringBond: RingBalance,
         ktonValue: KtonBalance,
	ktonBond: KtonBalance,
}
```

Reply:
And in this way, we may remove ```StakingBalance ``` enum, it seems not required anymore.

Reply:
For checking, ringBond must be larger than minimum, either, ktonBond must be larger than minimum. That is, it can be only proposal for ring or for kton, if the other one is zero.

Reply:
I don't agree with removing `StakingBalance` totally, because, in some logics, it simplify methods` logic practically.

I'll push a new version about keeping both `StakingBalance` and `Proposal<AccountId, RingBalance, KtonBalance>`, try to simplify the code as possible as it can be.